packages/@aws-cdk/aws-lambda/README.md

```diff
@@ -277,6 +277,42 @@ correct log retention period (never expire, by default).
 *Further note* that, if the log group already exists and the `logRetention` is not set, the custom resource will reset
 the log retention to never expire even if it was configured with a different value.
 
+### FileSystem Access
+
+You can configure a function to mount an Amazon Elastic File System (Amazon EFS) to a
+directory in your runtime environment with the `filesystems` property. To access Amaozn EFS
+from lambda function, the Amazon EFS access point will be required. 
+
+Amazon EFS _access points_ are application-specific entry points into an EFS file system that
+make it easier to manage application access to shared datasets. Access points can enforce a user
+identity, including the user's POSIX groups, for all file system requests that are made through
+the access point. Access points can also enforce a different root directory for the file system
+so that clients can only access data in the specified directory or its subdirectories. See
+[Working with Amazon EFS Access Points](https://docs.aws.amazon.com/efs/latest/ug/efs-access-points.html) for more details.
+
+The following sample enables the lambda function to mount the Amazon EFS filesystem via the `accessPoint` to `/mnt/msg` in the runtime environment.
+
+```ts
+const fileSystem = new efs.FileSystem(stack, 'Efs', { vpc });
+
+const accessPoint = fileSystem.addAccessPoint('AccessPoint');
+
+const fn = new lambda.Function(stack, 'MyLambda', {
+  code,
+  handler,
+  runtime,
+  vpc,
+  securityGroups: fileSystem.connections.securityGroups,
+  filesystems: [{
+    target: new lambda.EfsAccessPointTarget(accessPoint),
+    mountPath:  '/mnt/msg',
+  }],
+});
+
+fn.node.addDependency(accessPoint, fileSystem);
```

Comment:
Why is this needed? This should be implicit

Reply:
Actually the lambda function will need to depend on the `CfnMountTarget` resources which were created [under the hood](https://github.com/aws/aws-cdk/blob/75e6d255e8a6f3a45f0c65905e23bf058c8ced27/packages/%40aws-cdk/aws-efs/lib/efs-file-system.ts#L267) when we new `efs.FileSystem()`

From my previous tests we need at least

```
fn.node.addDependency(fileSystem);
```

Otherwise we'll encounter the mount targets not ready error. 

Did I missing anything?

Reply:
That’s okay, but this can be automatically added for the user, no?

Reply:
> That’s okay, but this can be automatically added for the user, no?

I am afraid not. From the `lambda.Function`'s perspective, all we have is only the `accessPoint` and there is no reference to the `efs.FileSystem` or mount targets from the given `accessPoint` so we can't do it in the lambda function. We probably can add another PR from the EFS constructs to expose those resources from the `accessPoint`. Does it make sense?

Reply:
I think `accessPoint.filesystem` is in order and it should be sufficient because if `CfnMountTarget` is a child of the FileSystem Construct, then the dependency will transitively apply. 

I think we can add it in this PR

Reply:
Hi @eladb 

I still found it challenging as in `efs.AccessPoint` the `fileSystem` construct property actually is `IFileSystem`

https://github.com/aws/aws-cdk/blob/83fd2aee6389d03cfe69260b996d7d09398bbf99/packages/%40aws-cdk/aws-efs/lib/access-point.ts#L108

so the `accessPoint.filesystem` exposed will actually `IFileSystem` and lambda.Function can't depend on it.

Comments?